dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -815,3 +876,129 @@ CLSID GetClsid(const winrt::Windows::ApplicationModel::AppExtensions::AppExtensi
+    PACKAGE_VERSION minVersion) noexcept try
+{
+    HANDLE hEventLog{ RegisterEventSourceW(nullptr, L"Windows App Runtime") };
+    RETURN_LAST_ERROR_IF_NULL(hEventLog);
+
+    const DWORD c_eventId{ static_cast<DWORD>(hrInitialize) };
+    PCWSTR message1{ L"Windows App Runtime" };
+    WCHAR message2[1024]{};
+    PCWSTR message2Format{ L"ERROR 0x%08X: Bootstrapper initialization failed while looking for version %hu.%hu%s (MSIX packages version >= %hu.%hu.%hu.%hu)" };
+    const UINT16 majorVersion{ HIWORD(majorMinorVersion) };
+    const UINT16 minorVersion{ LOWORD(majorMinorVersion) };
+    WCHAR formattedVersionTag[64]{};
+    if (versionTag && (versionTag[0] != L'\0'))
+    {
+        FAIL_FAST_IF_FAILED(StringCchPrintfW(formattedVersionTag, ARRAYSIZE(formattedVersionTag), L"-%s", versionTag));
+    }
+    FAIL_FAST_IF_FAILED(StringCchPrintfW(message2, ARRAYSIZE(message2), message2Format,
+                                         hrInitialize, majorVersion, minorVersion, formattedVersionTag,
+                                         minVersion.Major, minVersion.Minor, minVersion.Build, minVersion.Revision));
+    PCWSTR strings[2]{ message1, message2 };
+    LOG_IF_WIN32_BOOL_FALSE(ReportEventW(hEventLog, EVENTLOG_ERROR_TYPE, 0, c_eventId, nullptr, ARRAYSIZE(strings), 0, strings, nullptr));
+
+    DeregisterEventSource(hEventLog);
+
+    return S_OK;
+}
+CATCH_RETURN()
+
+HRESULT MddBootstrapInitialize_ShowUI(
+    UINT32 majorMinorVersion,
+    PCWSTR versionTag,
+    PACKAGE_VERSION minVersion)
+{
+    //TODO: Show this if ERROR_NO_MATCH vs simpler MessageBox("ERROR 0xN in Bootstrapper initialize", MB_OK) ?
+
+    // Get the message caption
+    PCWSTR caption{};
+    wil::unique_cotaskmem_string captionString;
+    WCHAR captionOnError[100]{};
+    try
+    {
+        PCWSTR executable{};
+        wil::unique_cotaskmem_string module;
+        auto hr{ LOG_IF_FAILED(wil::GetModuleFileNameW(nullptr, module)) };
+        if (SUCCEEDED(hr))
+        {
+            auto delimiter{ wcsrchr(module.get(), L'\\') };
+            if (delimiter)
+            {
+                executable = delimiter + 1;
+            }
+            else
+            {
+                executable = module.get();
+            }
+            PCWSTR captionSuffix{ L"This application could not be started" };
+            captionString = wil::str_printf<wil::unique_cotaskmem_string>(L"%s - %s", executable, captionSuffix);
+            caption = captionString.get();
+        }
+    }
+    catch (...)
+    {
+    }
+    if (!caption)
+    {
+        LOG_IF_FAILED(StringCchPrintfW(captionOnError, ARRAYSIZE(captionOnError),
+                                       L"<Process %d> - This application could not be started",
+                                       GetCurrentProcessId()));
+        caption = captionOnError;
+    }
+
+    // Get the message body
+    WCHAR text[1024]{};
+    PCWSTR textFormat{ L"This application requires the Windows App Runtime\n"
```

Comment:
Should this be in a localizable resource file? (e.g. `.rc`) Or is that for a later round w/ loc lens?

Reply:
> Should this be in a localizable resource file? (e.g. `.rc`) Or is that for a later round w/ loc lens?

Expected later, as a subsequent PR.

Text is segregated to facilitate eventual move to STRINGTABLE resources. Needs a bit more than simple STRINGTABLE lookup e.g. sprintf and %s, %hd, etc will need to change to FormatMessage() and %1 etc to properly support localization.